### ISSUE TITLE
Argument 1 passed to Spatie\MediaLibrary\Filesystem\DefaultFilesystem::getRemoteHeadersForFile() must be of the type string, resource given

### ISSUE THREAD

**Issue by sebestenyb** [spatie/laravel-medialibrary#546]
Hi

With a blank new project:

```
// composer.json
    "require": {
        "php": ">=5.6.4",
        "laravel/framework": "5.4.*",
        "laravel/tinker": "~1.0",
        "league/flysystem-aws-s3-v3": "^1.0",
        "spatie/laravel-medialibrary": "^5.3.2"
    },
```

```
// filesystem.php
        'media' => [
            'driver' => 's3',
            'key'    => env('AWS_KEY'),
            'secret' => env('AWS_SECRET'),
            'region' => env('AWS_REGION'),
            'bucket' => env('AWS_BUCKET'),
        ]
```

```
// User.php
use Illuminate\Notifications\Notifiable;
use Illuminate\Foundation\Auth\User as Authenticatable;
use Spatie\MediaLibrary\HasMedia\HasMediaTrait;
use Spatie\MediaLibrary\HasMedia\Interfaces\HasMedia;

class User extends Authenticatable implements HasMedia
{
    use Notifiable, HasMediaTrait;
```

```
// tinker
Psy Shell v0.8.2 (PHP 7.1.2 — cli) by Justin Hileman
>>> use App\User
=> null
>>> use Faker\Factory
=> null
>>> $faker = Factory::create();
=> Faker\Generator {#685}
>>> User::first()->addMediaFromUrl($faker->imageUrl())->toMediaLibrary();
TypeError: Argument 1 passed to Spatie\MediaLibrary\Filesystem\DefaultFilesystem::getRemoteHeadersForFile() must be of the type string, resource given, called in /Users/balazssebesteny/Sites/medialibrary-test/vendor/spatie/laravel-medialibrary/src/Filesystem/DefaultFilesystem.php on line 60
```

Cheers, Sebi

**Comment by freekmurze** [comment]
This has been fixed in `v5.3.3`

**Comment by sebestenyb** [comment]
!! Nice, you guys are fast !!
Thanks for all the effort 👍
